crates/trident/src/subsystems/extensions/release.rs

```diff
@@ -0,0 +1,306 @@
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{ensure, Context, Error};
+use log::debug;
+
+use osutils::osrelease::ExtensionRelease;
+use trident_api::config::Extension;
+
+use crate::subsystems::extensions::{
+    ExtensionData, ExtensionType, CONFEXT_EXTENSION_RELEASE_DIRECTORY, DEFAULT_CONFEXT_DIRECTORY,
+    DEFAULT_SYSEXT_DIRECTORY, SYSEXT_EXTENSION_RELEASE_DIRECTORY,
+};
+
+/// Helper function to extract information from extension-release file.
+pub(crate) fn read_extension_release(
+    mount_point: &Path,
+    curr_path: &Path,
+    ext: &Extension,
+    ext_type: &ExtensionType,
+) -> Result<ExtensionData, Error> {
+    debug!(
+        "Processing extension release file for extension image at '{}'",
+        ext.url
+    );
+
+    let sysext_release_dir = mount_point.join(SYSEXT_EXTENSION_RELEASE_DIRECTORY);
+    let confext_release_dir = mount_point.join(CONFEXT_EXTENSION_RELEASE_DIRECTORY);
+
+    // Get extension release file
+    let dir = match ext_type {
+        ExtensionType::Sysext => fs::read_dir(&sysext_release_dir).with_context(|| format!("Failed to find extension release directory '{SYSEXT_EXTENSION_RELEASE_DIRECTORY}' in image at '{}'", ext.url))?,
```

Comment:
nit nit nit nit: this is a really long line ... it is surprising that cargo doesn't reformat it.